zulipterminal/ui_tools/views.py

```diff
@@ -276,6 +276,16 @@ def update_streams(self, search_box: Any, new_text: str) -> None:
                 for stream in self.streams_btn_list.copy()
                 if new_text in stream.stream_name.lower()
             ]
+            streams_display.sort(key=lambda stream: stream.stream_name.lower())
+            # Give priority to the streams which start with the `new_text`.
+            streams_display.sort(
+                key=(lambda stream:
+                     stream.stream_name.lower().startswith(new_text)),
+                reverse=True)
+            # Order by `pin_to_top` status.
+            streams_display.sort(key=lambda stream: stream.stream_name in map(
+                lambda stream: stream[0], self.view.pinned_streams),
+                reverse=True)
```

Comment:
This code looks very familiar to the other? Is there a reason you didn't refactor it first?

Reply:
@neiljp I had thought of introducing `order_stream`, to sort the streams as per these criteria, but that required some pre-refactoring and structural changes in `update_streams` and `autocomplete_streams`.

Should I go ahead with this? Also, what could be the commit message for pre-refactoring commits like these (where some refactoring is required only for getting it ready for a 'new' feature)?

Reply:
Refactoring is generally fine to do first. Otherwise you can often end up duplicating and slightly altering code, only to shrink it in a refactoring commit afterwards.

Would it make sense to move this code into `match_stream`, or perhaps once you've rebased against the refactoring I merged from @kaustubh-nair this will be more straight-forward in any case?